packages/core/src/controllers/ChainController.ts

```diff
@@ -259,29 +309,32 @@ export const ChainController = {
   },
 
   resetAccount(chain?: Chain) {
-    const chainToWrite = state.multiChainEnabled ? chain : state.activeChain
+    const chainToWrite = chain
 
     if (!chainToWrite) {
       throw new Error('Chain is required to set account prop')
     }
 
-    this.setChainAccountData(chainToWrite, {
-      isConnected: false,
-      smartAccountDeployed: false,
-      currentTab: 0,
-      caipAddress: undefined,
-      address: undefined,
-      balance: undefined,
-      balanceSymbol: undefined,
-      profileName: undefined,
-      profileImage: undefined,
-      addressExplorerUrl: undefined,
-      tokenBalance: [],
-      connectedWalletInfo: undefined,
-      preferredAccountType: undefined,
-      socialProvider: undefined,
-      socialWindow: undefined,
-      farcasterUrl: undefined
-    })
+    this.setChainAccountData(
+      chainToWrite,
+      ref({
```

Comment:
Is this needed

Reply:
Since we are not setting these values to another objects' field, we need to use `ref` for non-primitive values (for objects/arrays).

Previously we were setting these values of the state of the following:

```js
const state = {
 isConnected: false,
 currentTab: 0,
 ...
}
```

Not it's assigned to this:

```js
const state = {
  evm: {
    networkState:{ // this will be updated at once
      isConnected: false,
      currentTab: 0,
    }
  }
  solana: {...}
 ...
}
```